libs/game/sprite.ts

```diff
@@ -939,6 +1083,90 @@ class Sprite extends sprites.BaseSprite {
         }
     }
 
+    setScaleCore(sx?: number, sy?: number, anchor?: ScaleAnchor, proportional?: boolean): void {
+        anchor = anchor || ScaleAnchor.Middle;
+
+        const hasSx = typeof sx === 'number';
+        const hasSy = typeof sy === 'number';
+
+        const oldW = this.width;
+        const oldH = this.height;
+
+        if (hasSx) {
+            const oldSx = this.sx;
+            this.sx = sx;
+            if (!hasSy && proportional) {
+                const ratio = sx / oldSx;
+                this.sy *= ratio;
+            }
+        }
+        if (hasSy) {
+            const oldSy = this.sy;
+            this.sy = sy;
+            if (!hasSx && proportional) {
+                const ratio = sy / oldSy;
+                this.sx *= ratio;
+            }
+        }
+
+        if (anchor & (ScaleAnchor.Left | ScaleAnchor.Right)) {
+            const newW = this.width;
+            const diff = newW - oldW;
+            const diffOver2 = (diff / 2) | 0;
+            if (anchor & ScaleAnchor.Left) { this.x += diffOver2; }
+            if (anchor & ScaleAnchor.Right) { this.x -= diffOver2; }
+        }
+        if (anchor & (ScaleAnchor.Top | ScaleAnchor.Bottom)) {
+            const newH = this.height;
+            const diff = newH - oldH;
+            const diffOver2 = (diff / 2) | 0;
+            if (anchor & ScaleAnchor.Top) { this.y += diffOver2; }
+            if (anchor & ScaleAnchor.Bottom) { this.y -= diffOver2; }
+        }
+    }
+
+    //% blockId=sprite_set_scale
+    //% block="set %sprite(mySprite) scale to $value, anchor on $anchor"
```

Comment:
oh, one last one, I think I might lean more towards `"set %sprite(mySprite) scale to $value anchored on $anchor"` as I don't recall any other blocks using punctuation as much, but that's just my opinion / if you feel it's better as is that's probably fine~